todo-app/src/TextField.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+
+export default class TextField extends React.Component {
+
+    render() {
+        if (this.props.multiline) {
+            return this.renderMultiLine();
+        } else if (this.props.singleline) { 
+            return this.renderSingleLine();
+        } else {
+            return this.renderDateForm();
+        }
+    };
+
+    renderSingleLine() {
+        return (
+            <div>
+                <input type="text" 
+                className="form"
+                value={this.props.value}
+                onChange={this.props.onChange}
+                />
+            </div>
+        );
+    };
+
+    renderDateForm() {
+        return (
+            <div>
+                <input type="date" 
+                className="date"
+                value={this.props.value}
+                onChange={this.props.onChange}
```

Comment:
If I enter '2018-01-27' in the field, is `onChange` called with a `Date` or with a `string`? And what would you want to get out of the field, a `Date` or a `string`?